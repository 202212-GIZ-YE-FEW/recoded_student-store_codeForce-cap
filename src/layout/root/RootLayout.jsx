import Head from "next/head"
import { serverSideTranslations } from "next-i18next/serverSideTranslations"

import Footer from "@/components/Footer"
import Navbar from "@/components/Navbar/Navbar"

export default function RootLayout({ children }) {
  // Put Header or Footer around the children element
  // Example
  return (
    <>
      <Head>
        <title>Student Store</title>
        <meta name='description' content='Student Store' />
        <meta name='viewport' content='width=device-width, initial-scale=1' />
        <link rel='icon' href='/images/tapLogo.png' />
      </Head>
      <Navbar />
      {children}
      <Footer />
    </>
  )
}

export async function getStaticProps({ locale }) {
  return {
    props: {
      ...(await serverSideTranslations(locale, ["common"])),
      // Will be passed to the page component as props
    },
  }
}
