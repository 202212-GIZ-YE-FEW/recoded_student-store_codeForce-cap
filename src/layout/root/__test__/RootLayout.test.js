import renderer from "react-test-renderer"
import RootLayout from "../RootLayout"

it("renders correctly", () => {
  const tree = renderer.create(<RootLayout />).toJSON()
  expect(tree).toMatchSnapshot()
})
