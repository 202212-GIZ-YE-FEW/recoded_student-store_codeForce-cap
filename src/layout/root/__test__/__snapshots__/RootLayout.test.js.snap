// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
[
  <header
    className="sticky top-0 z-50"
  >
    <div
      className="navbar m-auto px-10 md:flex md:flex-row py-4 justify-between gap-5 items-center bg-white transition-all 
      "
    >
      <a
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <img
          alt="logo"
          data-nimg="1"
          decoding="async"
          height={130}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fimages%2FLogo.png&w=384&q=75"
          srcSet="/_next/image?url=%2Fimages%2FLogo.png&w=256&q=75 1x, /_next/image?url=%2Fimages%2FLogo.png&w=384&q=75 2x"
          style={
            {
              "color": "transparent",
            }
          }
          width={130}
        />
      </a>
      <div
        className="flex items-end cursor-pointer absolute right-14 top-8 md:static order-2 md:hidden"
      >
        <div
          className="flex items-end"
          onClick={[Function]}
        >
          <svg
            className="text-2xl text-gray-700"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              {
                "color": undefined,
              }
            }
            version="1.1"
            viewBox="0 0 17 17"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g />
            <path
              d="M8.516 0c-4.687 0-8.5 3.813-8.5 8.5s3.813 8.5 8.5 8.5 8.5-3.813 8.5-8.5-3.814-8.5-8.5-8.5zM1.041 9h2.937c0.044 1.024 0.211 2.031 0.513 3h-2.603c-0.481-0.906-0.776-1.923-0.847-3zM3.978 8h-2.937c0.071-1.077 0.366-2.094 0.847-3h2.6c-0.301 0.969-0.467 1.976-0.51 3zM5.547 5h5.896c0.33 0.965 0.522 1.972 0.569 3h-7.034c0.046-1.028 0.239-2.035 0.569-3zM4.978 9h7.035c-0.049 1.028-0.241 2.035-0.572 3h-5.891c-0.331-0.965-0.524-1.972-0.572-3zM13.013 9h2.978c-0.071 1.077-0.366 2.094-0.847 3h-2.644c0.302-0.969 0.469-1.976 0.513-3zM13.013 8c-0.043-1.024-0.209-2.031-0.51-3h2.641c0.48 0.906 0.775 1.923 0.847 3h-2.978zM14.502 4h-2.354c-0.392-0.955-0.916-1.858-1.55-2.7 1.578 0.457 2.938 1.42 3.904 2.7zM9.074 1.028c0.824 0.897 1.484 1.9 1.972 2.972h-5.102c0.487-1.071 1.146-2.073 1.97-2.97 0.199-0.015 0.398-0.030 0.602-0.030 0.188 0 0.373 0.015 0.558 0.028zM6.383 1.313c-0.629 0.838-1.151 1.737-1.54 2.687h-2.314c0.955-1.267 2.297-2.224 3.854-2.687zM2.529 13h2.317c0.391 0.951 0.915 1.851 1.547 2.689-1.561-0.461-2.907-1.419-3.864-2.689zM7.926 15.97c-0.826-0.897-1.488-1.899-1.978-2.97h5.094c-0.49 1.072-1.152 2.075-1.979 2.972-0.181 0.013-0.363 0.028-0.547 0.028-0.2 0-0.395-0.015-0.59-0.030zM10.587 15.703c0.636-0.842 1.164-1.747 1.557-2.703h2.358c-0.968 1.283-2.332 2.247-3.915 2.703z"
            />
          </svg>
          <svg
            className="text-xl text-violet-700"
            fill="none"
            height="1em"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
              stroke="none"
            />
            <path
              d="M17 13v-6l-5 4l-5 -4v6l5 4z"
            />
          </svg>
        </div>
        <div
          className="relative z-50 "
        >
          <div
            className="absolute left-[-100px] top-2 bg-white capitalize border border-solid border-violet-600 hidden"
          >
            <div
              className="my-2 px-5 py-2 hover:bg-gray-200 transition-all duration-500"
            >
              english
            </div>
            <div
              className="my-2 px-5 py-2 hover:bg-gray-200 transition-all duration-500"
            >
              arabic
            </div>
          </div>
        </div>
      </div>
      <svg
        className="absolute right-6 top-8 text-2xl cursor-pointer md:hidden show"
        fill="none"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 15 15"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M1.5 3C1.22386 3 1 3.22386 1 3.5C1 3.77614 1.22386 4 1.5 4H13.5C13.7761 4 14 3.77614 14 3.5C14 3.22386 13.7761 3 13.5 3H1.5ZM1 7.5C1 7.22386 1.22386 7 1.5 7H13.5C13.7761 7 14 7.22386 14 7.5C14 7.77614 13.7761 8 13.5 8H1.5C1.22386 8 1 7.77614 1 7.5ZM1 11.5C1 11.2239 1.22386 11 1.5 11H13.5C13.7761 11 14 11.2239 14 11.5C14 11.7761 13.7761 12 13.5 12H1.5C1.22386 12 1 11.7761 1 11.5Z"
          fill="currentColor"
          fillRule="evenodd"
        />
      </svg>
      <svg
        className="absolute right-6 top-8 text-2xl cursor-pointer md:hidden hidden"
        fill="currentColor"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
        />
      </svg>
      <div
        className="flex-1 md:static flex items-center flex-col gap-5 md:flex-row md:gap-2 md:z-auto absolute  top-[-1000px] left-0  w-full  md:w-auto bg-white p-5 md:p-0 transition-all duration-500 ease-in-out"
      >
        <div
          className="flex flex-col md:flex-row gap-5"
        >
          <a
            className="relative mx-4 capitalize text-sm hover:text-violet-800   transition-all duration-1000 before:absolute before:-bottom-2 before:left-0  before:h-[1px]  before:w-0 before:opacity-0 before:bg-violet-700 before:transition-all before:duration-500 hover:before:w-full hover:before:opacity-100"
            href="/donation"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            Donation
          </a>
          <a
            className="relative mx-4 capitalize text-sm  hover:text-violet-800   transition-all duration-1000 before:absolute before:-bottom-2 before:left-0  before:h-[1px]  before:w-0 before:opacity-0 before:bg-violet-700 before:transition-all before:duration-500 hover:before:w-full hover:before:opacity-100"
            href="/about"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            about us
          </a>
        </div>
        <div
          className=" flex-1 border flex items-center py-2 px-4 rounded-3xl"
        >
          <input
            className="search input w-full text-sm placeholder:text-gray-300 placeholder:capitalize text-gray-600
              "
            placeholder="search here"
            type="text"
          />
          <div
            className=""
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
              />
            </svg>
          </div>
        </div>
        <div
          className="md:flex items-end cursor-pointer hidden"
        >
          <div
            className="flex items-end"
            onClick={[Function]}
          >
            <svg
              className="text-2xl text-gray-700"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              version="1.1"
              viewBox="0 0 17 17"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g />
              <path
                d="M8.516 0c-4.687 0-8.5 3.813-8.5 8.5s3.813 8.5 8.5 8.5 8.5-3.813 8.5-8.5-3.814-8.5-8.5-8.5zM1.041 9h2.937c0.044 1.024 0.211 2.031 0.513 3h-2.603c-0.481-0.906-0.776-1.923-0.847-3zM3.978 8h-2.937c0.071-1.077 0.366-2.094 0.847-3h2.6c-0.301 0.969-0.467 1.976-0.51 3zM5.547 5h5.896c0.33 0.965 0.522 1.972 0.569 3h-7.034c0.046-1.028 0.239-2.035 0.569-3zM4.978 9h7.035c-0.049 1.028-0.241 2.035-0.572 3h-5.891c-0.331-0.965-0.524-1.972-0.572-3zM13.013 9h2.978c-0.071 1.077-0.366 2.094-0.847 3h-2.644c0.302-0.969 0.469-1.976 0.513-3zM13.013 8c-0.043-1.024-0.209-2.031-0.51-3h2.641c0.48 0.906 0.775 1.923 0.847 3h-2.978zM14.502 4h-2.354c-0.392-0.955-0.916-1.858-1.55-2.7 1.578 0.457 2.938 1.42 3.904 2.7zM9.074 1.028c0.824 0.897 1.484 1.9 1.972 2.972h-5.102c0.487-1.071 1.146-2.073 1.97-2.97 0.199-0.015 0.398-0.030 0.602-0.030 0.188 0 0.373 0.015 0.558 0.028zM6.383 1.313c-0.629 0.838-1.151 1.737-1.54 2.687h-2.314c0.955-1.267 2.297-2.224 3.854-2.687zM2.529 13h2.317c0.391 0.951 0.915 1.851 1.547 2.689-1.561-0.461-2.907-1.419-3.864-2.689zM7.926 15.97c-0.826-0.897-1.488-1.899-1.978-2.97h5.094c-0.49 1.072-1.152 2.075-1.979 2.972-0.181 0.013-0.363 0.028-0.547 0.028-0.2 0-0.395-0.015-0.59-0.030zM10.587 15.703c0.636-0.842 1.164-1.747 1.557-2.703h2.358c-0.968 1.283-2.332 2.247-3.915 2.703z"
              />
            </svg>
            <svg
              className="text-xl text-violet-700"
              fill="none"
              height="1em"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
                stroke="none"
              />
              <path
                d="M17 13v-6l-5 4l-5 -4v6l5 4z"
              />
            </svg>
          </div>
          <div
            className="relative"
          >
            <div
              className="absolute left-[-100px] top-5 bg-white capitalize hidden"
            >
              <div
                className="my-2 px-5 py-2 hover:bg-gray-200 transition-all duration-500"
              >
                english
              </div>
              <div
                className="my-2 px-5 py-2 hover:bg-gray-200 transition-all duration-500"
              >
                arabic
              </div>
            </div>
          </div>
        </div>
        <div
          className="flex  justify-between items-center gap-5 flex-col md:flex-row "
        >
          <div
            className="flex gap-3 capitalize flex-col md:flex-row"
          >
            <a
              href="/signin"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
            >
              <div
                className="bg-purple-light py-2 px-5 text-white rounded-3xl text-sm hover:bg-violet-800 transition-all cursor-pointer"
              >
                sign in
              </div>
            </a>
            <a
              href="/listing"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
            >
              <div
                className="bg-purple-light py-2 px-5 text-white rounded-3xl text-sm hover:bg-violet-800 transition-all cursor-pointer"
              >
                sell items
              </div>
            </a>
          </div>
          <div
            className="text-2xl text-orange-600 hover:text-orange-700 transition"
          >
            <a
              href="/cart"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
            >
               
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  {
                    "color": undefined,
                  }
                }
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clipRule="evenodd"
                  d="M3.172 5.172a4 4 0 015.656 0L10 6.343l1.172-1.171a4 4 0 115.656 5.656L10 17.657l-6.828-6.829a4 4 0 010-5.656z"
                  fillRule="evenodd"
                />
              </svg>
               
            </a>
          </div>
        </div>
      </div>
    </div>
  </header>,
  <footer>
    <div
      className="relative lg:flex lg:justify-evenly bg-[#32314C] w-full h-full break-all mt-auto pb-32"
    >
      <div
        className="ml-5 pt-5"
      >
        <a
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <img
            alt="logo"
            data-nimg="1"
            decoding="async"
            height={250}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/images/logo-footer.svg"
            style={
              {
                "color": "transparent",
              }
            }
            width={350}
          />
        </a>
        <div
          className="absolute bottom-16 left-5 lg:left-40 lg:bottom-1"
        >
          <a
            href="#"
          >
            <img
              alt="Google Play"
              className="box-content p-2 pt-6 md:w-32 lg:w-48"
              data-nimg="1"
              decoding="async"
              height={49}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/images/play-google.svg"
              style={
                {
                  "color": "transparent",
                }
              }
              width={175}
            />
          </a>
          <a
            href="#"
          >
            <img
              alt="App Store"
              className="box-content p-2 md:w-32 lg:w-48"
              data-nimg="1"
              decoding="async"
              height={49}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/images/app-store.svg"
              style={
                {
                  "color": "transparent",
                }
              }
              width={175}
            />
          </a>
        </div>
      </div>
      <div
        className="pl-14 pt-5 md:pt-14"
      >
        <ul
          className="font-poppins text-white font-bold md:text-1xl lg:text-2xl"
        >
          <li>
            <a
              className="hover:underline"
              href="/about"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
            >
              About Us
            </a>
          </li>
          <li>
            <h2
              className="pt-4"
            >
              Contact Us:
            </h2>
            <ul
              className="text-gray-600 dark:text-gray-400 font-medium"
            >
              <li
                className="pt-4"
              >
                (555)678-9012
              </li>
              <li
                className="pt-2"
              >
                contact@studentstore.com
              </li>
            </ul>
          </li>
        </ul>
      </div>
      <div
        className="pl-14 pt-10 md:pt-14"
      >
        <ul
          className="font-poppins text-white font-bold md:text-1xl lg:text-2xl"
        >
          <li
            className="mb-4"
          >
            <a
              className="hover:underline "
              href="/signup"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
            >
              Register for free
            </a>
          </li>
          <li
            className="mb-4"
          >
            <a
              className="hover:underline"
              href="/listing"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
            >
              Start Selling
            </a>
          </li>
          <li
            className="mb-4"
          >
            <a
              className="hover:underline"
              href="/"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
            >
              Buy Products
            </a>
          </li>
        </ul>
        <div
          className="absolute md:flex justify-between bottom-64 right-0 lg:right-96 lg:top-64"
        >
          <a
            href="#"
          >
            <img
              alt="Facebook"
              className="md:w-15 lg:w-15 m-5"
              data-nimg="1"
              decoding="async"
              height={70}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/images/facebook.svg"
              style={
                {
                  "color": "transparent",
                }
              }
              width={60}
            />
          </a>
          <a
            href="#"
          >
            <img
              alt="Twitter"
              className="md:w-15 lg:w-15 m-5"
              data-nimg="1"
              decoding="async"
              height={70}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/images/twitter.svg"
              style={
                {
                  "color": "transparent",
                }
              }
              width={60}
            />
          </a>
          <a
            href="#"
          >
            <img
              alt="Instagram"
              className="md:w-15 lg:w-15 m-5"
              data-nimg="1"
              decoding="async"
              height={70}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/images/instagram.svg"
              style={
                {
                  "color": "transparent",
                }
              }
              width={60}
            />
          </a>
        </div>
      </div>
      <div
        className="flex justify-end h-36 pt-32 lg:pt-14"
      >
        <h2
          className="m-0 font-poppins text-white font-bold md:text-2xl lg:text-2xl pt-2 lg:pt-7"
        >
          Scroll Up!
        </h2>
        <button
          onClick={[Function]}
        >
          <img
            alt="UP"
            className="w-10 ml-2 mr-5 lg:mr-0"
            data-nimg="1"
            decoding="async"
            height={35}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/images/up-arrow.svg"
            style={
              {
                "color": "transparent",
              }
            }
            width={45}
          />
        </button>
      </div>
    </div>
  </footer>,
]
`;
